sources/Interop/Windows/Windows.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace TerraFX.Interop
+{
+    public static unsafe partial class Windows
+    {
+        /// <summary>Raised whenever a native library is loaded by TerraFX.Interop.Windows. Handlers can be added to this event to customize how libraries are loaded, and they will be used first whenever a new native library is being resolved.</summary>
+        public static event DllImportResolver? ResolveLibrary;
+
+        static Windows()
+        {
+            NativeLibrary.SetDllImportResolver(Assembly.GetExecutingAssembly(), OnDllImport);
```

Comment:
You don't have to call this until there's an event subscription on `ResolveLibrary`, e.g.

`private static DllImportResolver? resolveLibrary;
public static DllImportResolver? ResolverLibrary
{
    add
    {
        ... do it here ...
    }
}
```

This helps ensure this code path isn't executed unless needed. And, this would certainly cause problems if anyone decides to do IL merging and also needs to call SetDllImportResolver() on the assembly that TerraFX is being merged into.

Reply:
This has to be done up front because `SetDllImportResolver` needs to be done before the first P/Invoke call happens because a given library is resolved once and you can only have a single hook for a given assembly. This means if I don't set it up and first thing, it causes downstream issues if users try to set up their own or if I ever need to add default resolution.

Default resolution logic is required for some assemblies, like vulkan, because the name differs per OS, its also theoretically needed for cases like x3daudio or d3dcompiler to help avoid versioning issues because the assembly name can change across versions.